zomes/hc_zome_trust_atom/src/trust_atom.rs

```diff
@@ -109,14 +115,31 @@ fn trust_atom_link_tag(link_direction: &LinkDirection, mut chunks: Vec<&str>) ->
   let content = chunks.remove(0);
   link_tag_bytes.extend_from_slice(content.as_bytes());
 
-  for chunk in &chunks {
+  for chunk in chunks {
     link_tag_bytes.extend_from_slice(&UNICODE_NUL_BYTES);
     link_tag_bytes.extend_from_slice(chunk.as_bytes());
   }
 
   LinkTag(link_tag_bytes)
 }
 
+fn gen_bucket() -> ExternResult<String> {
```

Comment:
I suggest create_bucket for the name.  

For algo, I suggest:

- get random_bytes(4)
- This gives you 4 ints each 0-255
- from these, derive a single int, in range 0 - 255^4
- take this % 10^9
- Now you have an int in range 0 - 999999999
- Format to a string, adding leading zeros if needed
- for this You can use format! With the right format args
- done 😎

Reply:
Also suggest we break this into 2 functions— one takes a vec of bytes, and does all the useful work and can be unit tested, the other calls hdk function. 

Then add a unit test which calls the first one 1000 times.